Compile bug:

### Git commit

I tried to build it, but I got an error.
Here are the commands I ran:

```
git clone https://github.com/ggerganov/llama.cpp
cd llama.cpp
mkdir build
cd build
cmake ..
make
```




### Operating systems

Linux

### GGML backends

CPU, CUDA


### Relevant log output

```shell
(base) sagemaker-user@default:~$ git clone https://github.com/ggerganov/llama.cpp
cd llama.cpp
mkdir build
cd build
cmake ..
make
Cloning into 'llama.cpp'...
remote: Enumerating objects: 67625, done.
remote: Counting objects: 100% (222/222), done.
remote: Compressing objects: 100% (178/178), done.
remote: Total 67625 (delta 151), reused 46 (delta 44), pack-reused 67403 (from 4)
Receiving objects: 100% (67625/67625), 196.38 MiB | 64.89 MiB/s, done.
Resolving deltas: 100% (48788/48788), done.
-- The C compiler identification is GNU 13.3.0
-- The CXX compiler identification is GNU 13.3.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /opt/conda/bin/x86_64-conda-linux-gnu-cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /opt/conda/bin/x86_64-conda-linux-gnu-c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMAKE_BUILD_TYPE=Release
-- Found Git: /usr/bin/git (found version "2.34.1") 
-- The ASM compiler identification is GNU
-- Found assembler: /opt/conda/bin/x86_64-conda-linux-gnu-cc
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Failed
-- Check if compiler accepts -pthread
-- Check if compiler accepts -pthread - yes
-- Found Threads: TRUE  
-- Warning: ccache not found - consider installing it for faster compilation or disable this warning with GGML_CCACHE=OFF
-- CMAKE_SYSTEM_PROCESSOR: x86_64
-- GGML_SYSTEM_ARCH: x86
-- Including CPU backend
-- Found OpenMP_C: -fopenmp (found version "4.5") 
-- Found OpenMP_CXX: -fopenmp (found version "4.5") 
-- Found OpenMP: TRUE (found version "4.5")  
-- x86 detected
-- Adding CPU backend variant ggml-cpu: -march=native 
-- ggml version: 0.9.4
-- ggml commit:  0cfb19166
-- Found CURL: /opt/conda/lib/libcurl.so (found version "8.13.0")  
-- Configuring done
-- Generating done
-- Build files have been written to: /home/sagemaker-user/llama.cpp/build
[  1%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml.c.o
[  1%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml.cpp.o
[  1%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml-alloc.c.o
[  1%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml-backend.cpp.o
[  2%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml-opt.cpp.o
[  2%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/ggml-threading.cpp.o
[  2%] Building C object ggml/src/CMakeFiles/ggml-base.dir/ggml-quants.c.o
[  3%] Building CXX object ggml/src/CMakeFiles/ggml-base.dir/gguf.cpp.o
[  3%] Linking CXX shared library ../../bin/libggml-base.so
[  3%] Built target ggml-base
[  3%] Building C object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ggml-cpu.c.o
[  3%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ggml-cpu.cpp.o
[  4%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/repack.cpp.o
[  4%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/hbm.cpp.o
[  4%] Building C object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/quants.c.o
[  5%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/traits.cpp.o
[  5%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/amx/amx.cpp.o
[  5%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/amx/mmq.cpp.o
[  5%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/binary-ops.cpp.o
[  6%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/unary-ops.cpp.o
[  6%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/vec.cpp.o
[  6%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/ops.cpp.o
[  7%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/llamafile/sgemm.cpp.o
[  7%] Building C object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/arch/x86/quants.c.o
[  7%] Building CXX object ggml/src/CMakeFiles/ggml-cpu.dir/ggml-cpu/arch/x86/repack.cpp.o
[  7%] Linking CXX shared library ../../bin/libggml-cpu.so
[  7%] Built target ggml-cpu
[  7%] Building CXX object ggml/src/CMakeFiles/ggml.dir/ggml-backend-reg.cpp.o
[  7%] Linking CXX shared library ../../bin/libggml.so
[  7%] Built target ggml
[  8%] Building CXX object src/CMakeFiles/llama.dir/llama.cpp.o
[  8%] Building CXX object src/CMakeFiles/llama.dir/llama-adapter.cpp.o
[  8%] Building CXX object src/CMakeFiles/llama.dir/llama-arch.cpp.o
[  9%] Building CXX object src/CMakeFiles/llama.dir/llama-batch.cpp.o
[  9%] Building CXX object src/CMakeFiles/llama.dir/llama-chat.cpp.o
[  9%] Building CXX object src/CMakeFiles/llama.dir/llama-context.cpp.o
[  9%] Building CXX object src/CMakeFiles/llama.dir/llama-cparams.cpp.o
[ 10%] Building CXX object src/CMakeFiles/llama.dir/llama-grammar.cpp.o
[ 10%] Building CXX object src/CMakeFiles/llama.dir/llama-graph.cpp.o
[ 10%] Building CXX object src/CMakeFiles/llama.dir/llama-hparams.cpp.o
[ 11%] Building CXX object src/CMakeFiles/llama.dir/llama-impl.cpp.o
[ 11%] Building CXX object src/CMakeFiles/llama.dir/llama-io.cpp.o
[ 11%] Building CXX object src/CMakeFiles/llama.dir/llama-kv-cache.cpp.o
[ 11%] Building CXX object src/CMakeFiles/llama.dir/llama-kv-cache-iswa.cpp.o
[ 12%] Building CXX object src/CMakeFiles/llama.dir/llama-memory.cpp.o
[ 12%] Building CXX object src/CMakeFiles/llama.dir/llama-memory-hybrid.cpp.o
[ 12%] Building CXX object src/CMakeFiles/llama.dir/llama-memory-recurrent.cpp.o
[ 13%] Building CXX object src/CMakeFiles/llama.dir/llama-mmap.cpp.o
[ 13%] Building CXX object src/CMakeFiles/llama.dir/llama-model-loader.cpp.o
[ 13%] Building CXX object src/CMakeFiles/llama.dir/llama-model-saver.cpp.o
[ 13%] Building CXX object src/CMakeFiles/llama.dir/llama-model.cpp.o
[ 14%] Building CXX object src/CMakeFiles/llama.dir/llama-quant.cpp.o
[ 14%] Building CXX object src/CMakeFiles/llama.dir/llama-sampling.cpp.o
[ 14%] Building CXX object src/CMakeFiles/llama.dir/llama-vocab.cpp.o
[ 15%] Building CXX object src/CMakeFiles/llama.dir/unicode-data.cpp.o
[ 15%] Building CXX object src/CMakeFiles/llama.dir/unicode.cpp.o
[ 15%] Building CXX object src/CMakeFiles/llama.dir/models/apertus.cpp.o
[ 15%] Building CXX object src/CMakeFiles/llama.dir/models/arcee.cpp.o
[ 16%] Building CXX object src/CMakeFiles/llama.dir/models/arctic.cpp.o
[ 16%] Building CXX object src/CMakeFiles/llama.dir/models/arwkv7.cpp.o
[ 16%] Building CXX object src/CMakeFiles/llama.dir/models/baichuan.cpp.o
[ 17%] Building CXX object src/CMakeFiles/llama.dir/models/bailingmoe.cpp.o
[ 17%] Building CXX object src/CMakeFiles/llama.dir/models/bailingmoe2.cpp.o
[ 17%] Building CXX object src/CMakeFiles/llama.dir/models/bert.cpp.o
[ 17%] Building CXX object src/CMakeFiles/llama.dir/models/bitnet.cpp.o
[ 18%] Building CXX object src/CMakeFiles/llama.dir/models/bloom.cpp.o
[ 18%] Building CXX object src/CMakeFiles/llama.dir/models/chameleon.cpp.o
[ 18%] Building CXX object src/CMakeFiles/llama.dir/models/chatglm.cpp.o
[ 19%] Building CXX object src/CMakeFiles/llama.dir/models/codeshell.cpp.o
[ 19%] Building CXX object src/CMakeFiles/llama.dir/models/cogvlm.cpp.o
[ 19%] Building CXX object src/CMakeFiles/llama.dir/models/cohere2-iswa.cpp.o
[ 20%] Building CXX object src/CMakeFiles/llama.dir/models/command-r.cpp.o
[ 20%] Building CXX object src/CMakeFiles/llama.dir/models/dbrx.cpp.o
[ 20%] Building CXX object src/CMakeFiles/llama.dir/models/deci.cpp.o
[ 20%] Building CXX object src/CMakeFiles/llama.dir/models/deepseek.cpp.o
[ 21%] Building CXX object src/CMakeFiles/llama.dir/models/deepseek2.cpp.o
[ 21%] Building CXX object src/CMakeFiles/llama.dir/models/dots1.cpp.o
[ 21%] Building CXX object src/CMakeFiles/llama.dir/models/dream.cpp.o
[ 22%] Building CXX object src/CMakeFiles/llama.dir/models/ernie4-5-moe.cpp.o
[ 22%] Building CXX object src/CMakeFiles/llama.dir/models/ernie4-5.cpp.o
[ 22%] Building CXX object src/CMakeFiles/llama.dir/models/exaone.cpp.o
[ 22%] Building CXX object src/CMakeFiles/llama.dir/models/exaone4.cpp.o
[ 23%] Building CXX object src/CMakeFiles/llama.dir/models/falcon-h1.cpp.o
[ 23%] Building CXX object src/CMakeFiles/llama.dir/models/falcon.cpp.o
[ 23%] Building CXX object src/CMakeFiles/llama.dir/models/gemma-embedding.cpp.o
[ 24%] Building CXX object src/CMakeFiles/llama.dir/models/gemma.cpp.o
[ 24%] Building CXX object src/CMakeFiles/llama.dir/models/gemma2-iswa.cpp.o
[ 24%] Building CXX object src/CMakeFiles/llama.dir/models/gemma3-iswa.cpp.o
[ 24%] Building CXX object src/CMakeFiles/llama.dir/models/gemma3n-iswa.cpp.o
[ 25%] Building CXX object src/CMakeFiles/llama.dir/models/glm4-moe.cpp.o
[ 25%] Building CXX object src/CMakeFiles/llama.dir/models/glm4.cpp.o
[ 25%] Building CXX object src/CMakeFiles/llama.dir/models/gpt2.cpp.o
[ 26%] Building CXX object src/CMakeFiles/llama.dir/models/gptneox.cpp.o
[ 26%] Building CXX object src/CMakeFiles/llama.dir/models/granite-hybrid.cpp.o
[ 26%] Building CXX object src/CMakeFiles/llama.dir/models/granite.cpp.o
[ 26%] Building CXX object src/CMakeFiles/llama.dir/models/grok.cpp.o
[ 27%] Building CXX object src/CMakeFiles/llama.dir/models/grovemoe.cpp.o
[ 27%] Building CXX object src/CMakeFiles/llama.dir/models/hunyuan-dense.cpp.o
[ 27%] Building CXX object src/CMakeFiles/llama.dir/models/hunyuan-moe.cpp.o
[ 28%] Building CXX object src/CMakeFiles/llama.dir/models/internlm2.cpp.o
[ 28%] Building CXX object src/CMakeFiles/llama.dir/models/jais.cpp.o
[ 28%] Building CXX object src/CMakeFiles/llama.dir/models/jamba.cpp.o
[ 28%] Building CXX object src/CMakeFiles/llama.dir/models/lfm2.cpp.o
[ 29%] Building CXX object src/CMakeFiles/llama.dir/models/llada-moe.cpp.o
[ 29%] Building CXX object src/CMakeFiles/llama.dir/models/llada.cpp.o
[ 29%] Building CXX object src/CMakeFiles/llama.dir/models/llama-iswa.cpp.o
[ 30%] Building CXX object src/CMakeFiles/llama.dir/models/llama.cpp.o
[ 30%] Building CXX object src/CMakeFiles/llama.dir/models/mamba.cpp.o
[ 30%] Building CXX object src/CMakeFiles/llama.dir/models/minicpm3.cpp.o
[ 30%] Building CXX object src/CMakeFiles/llama.dir/models/minimax-m2.cpp.o
[ 31%] Building CXX object src/CMakeFiles/llama.dir/models/mpt.cpp.o
[ 31%] Building CXX object src/CMakeFiles/llama.dir/models/nemotron-h.cpp.o
[ 31%] Building CXX object src/CMakeFiles/llama.dir/models/nemotron.cpp.o
[ 32%] Building CXX object src/CMakeFiles/llama.dir/models/neo-bert.cpp.o
[ 32%] Building CXX object src/CMakeFiles/llama.dir/models/olmo.cpp.o
[ 32%] Building CXX object src/CMakeFiles/llama.dir/models/olmo2.cpp.o
[ 32%] Building CXX object src/CMakeFiles/llama.dir/models/olmoe.cpp.o
[ 33%] Building CXX object src/CMakeFiles/llama.dir/models/openai-moe-iswa.cpp.o
[ 33%] Building CXX object src/CMakeFiles/llama.dir/models/openelm.cpp.o
[ 33%] Building CXX object src/CMakeFiles/llama.dir/models/orion.cpp.o
[ 34%] Building CXX object src/CMakeFiles/llama.dir/models/pangu-embedded.cpp.o
[ 34%] Building CXX object src/CMakeFiles/llama.dir/models/phi2.cpp.o
[ 34%] Building CXX object src/CMakeFiles/llama.dir/models/phi3.cpp.o
[ 34%] Building CXX object src/CMakeFiles/llama.dir/models/plamo.cpp.o
[ 35%] Building CXX object src/CMakeFiles/llama.dir/models/plamo2.cpp.o
[ 35%] Building CXX object src/CMakeFiles/llama.dir/models/plm.cpp.o
[ 35%] Building CXX object src/CMakeFiles/llama.dir/models/qwen.cpp.o
[ 36%] Building CXX object src/CMakeFiles/llama.dir/models/qwen2.cpp.o
[ 36%] Building CXX object src/CMakeFiles/llama.dir/models/qwen2moe.cpp.o
[ 36%] Building CXX object src/CMakeFiles/llama.dir/models/qwen2vl.cpp.o
[ 36%] Building CXX object src/CMakeFiles/llama.dir/models/qwen3.cpp.o
[ 37%] Building CXX object src/CMakeFiles/llama.dir/models/qwen3vl.cpp.o
[ 37%] Building CXX object src/CMakeFiles/llama.dir/models/qwen3vl-moe.cpp.o
[ 37%] Building CXX object src/CMakeFiles/llama.dir/models/qwen3moe.cpp.o
[ 38%] Building CXX object src/CMakeFiles/llama.dir/models/refact.cpp.o
[ 38%] Building CXX object src/CMakeFiles/llama.dir/models/rwkv6-base.cpp.o
[ 38%] Building CXX object src/CMakeFiles/llama.dir/models/rwkv6.cpp.o
[ 38%] Building CXX object src/CMakeFiles/llama.dir/models/rwkv6qwen2.cpp.o
[ 39%] Building CXX object src/CMakeFiles/llama.dir/models/rwkv7-base.cpp.o
[ 39%] Building CXX object src/CMakeFiles/llama.dir/models/rwkv7.cpp.o
[ 39%] Building CXX object src/CMakeFiles/llama.dir/models/seed-oss.cpp.o
[ 40%] Building CXX object src/CMakeFiles/llama.dir/models/smallthinker.cpp.o
[ 40%] Building CXX object src/CMakeFiles/llama.dir/models/smollm3.cpp.o
[ 40%] Building CXX object src/CMakeFiles/llama.dir/models/stablelm.cpp.o
[ 40%] Building CXX object src/CMakeFiles/llama.dir/models/starcoder.cpp.o
[ 41%] Building CXX object src/CMakeFiles/llama.dir/models/starcoder2.cpp.o
[ 41%] Building CXX object src/CMakeFiles/llama.dir/models/t5-dec.cpp.o
[ 41%] Building CXX object src/CMakeFiles/llama.dir/models/t5-enc.cpp.o
[ 42%] Building CXX object src/CMakeFiles/llama.dir/models/wavtokenizer-dec.cpp.o
[ 42%] Building CXX object src/CMakeFiles/llama.dir/models/xverse.cpp.o
[ 42%] Building CXX object src/CMakeFiles/llama.dir/models/graph-context-mamba.cpp.o
[ 42%] Linking CXX shared library ../bin/libllama.so
[ 42%] Built target llama
[ 42%] Building CXX object common/CMakeFiles/build_info.dir/build-info.cpp.o
[ 42%] Built target build_info
[ 42%] Building CXX object common/CMakeFiles/common.dir/arg.cpp.o
[ 42%] Building CXX object common/CMakeFiles/common.dir/chat-parser.cpp.o
[ 43%] Building CXX object common/CMakeFiles/common.dir/chat.cpp.o
[ 43%] Building CXX object common/CMakeFiles/common.dir/common.cpp.o
[ 43%] Building CXX object common/CMakeFiles/common.dir/console.cpp.o
[ 44%] Building CXX object common/CMakeFiles/common.dir/download.cpp.o
[ 44%] Building CXX object common/CMakeFiles/common.dir/json-partial.cpp.o
[ 44%] Building CXX object common/CMakeFiles/common.dir/json-schema-to-grammar.cpp.o
[ 44%] Building CXX object common/CMakeFiles/common.dir/llguidance.cpp.o
[ 45%] Building CXX object common/CMakeFiles/common.dir/log.cpp.o
[ 45%] Building CXX object common/CMakeFiles/common.dir/ngram-cache.cpp.o
[ 45%] Building CXX object common/CMakeFiles/common.dir/regex-partial.cpp.o
[ 46%] Building CXX object common/CMakeFiles/common.dir/sampling.cpp.o
[ 46%] Building CXX object common/CMakeFiles/common.dir/speculative.cpp.o
[ 46%] Linking CXX static library libcommon.a
[ 46%] Built target common
[ 46%] Building CXX object vendor/cpp-httplib/CMakeFiles/cpp-httplib.dir/httplib.cpp.o
[ 47%] Linking CXX static library libcpp-httplib.a
[ 47%] Built target cpp-httplib
[ 48%] Building CXX object tests/CMakeFiles/test-tokenizer-0.dir/test-tokenizer-0.cpp.o
[ 48%] Linking CXX executable ../bin/test-tokenizer-0
[ 48%] Built target test-tokenizer-0
[ 48%] Building CXX object tests/CMakeFiles/test-sampling.dir/test-sampling.cpp.o
[ 48%] Building CXX object tests/CMakeFiles/test-sampling.dir/get-model.cpp.o
[ 49%] Linking CXX executable ../bin/test-sampling
[ 49%] Built target test-sampling
[ 49%] Building CXX object tests/CMakeFiles/test-grammar-parser.dir/test-grammar-parser.cpp.o
[ 49%] Building CXX object tests/CMakeFiles/test-grammar-parser.dir/get-model.cpp.o
[ 50%] Linking CXX executable ../bin/test-grammar-parser
[ 50%] Built target test-grammar-parser
[ 50%] Building CXX object tests/CMakeFiles/test-grammar-integration.dir/test-grammar-integration.cpp.o
[ 51%] Building CXX object tests/CMakeFiles/test-grammar-integration.dir/get-model.cpp.o
[ 51%] Linking CXX executable ../bin/test-grammar-integration
[ 51%] Built target test-grammar-integration
[ 52%] Building CXX object tests/CMakeFiles/test-llama-grammar.dir/test-llama-grammar.cpp.o
[ 52%] Building CXX object tests/CMakeFiles/test-llama-grammar.dir/get-model.cpp.o
[ 52%] Linking CXX executable ../bin/test-llama-grammar
[ 52%] Built target test-llama-grammar
[ 52%] Building CXX object tests/CMakeFiles/test-chat.dir/test-chat.cpp.o
[ 53%] Building CXX object tests/CMakeFiles/test-chat.dir/get-model.cpp.o
[ 53%] Linking CXX executable ../bin/test-chat
[ 53%] Built target test-chat
[ 53%] Building CXX object tests/CMakeFiles/test-json-schema-to-grammar.dir/test-json-schema-to-grammar.cpp.o
[ 53%] Building CXX object tests/CMakeFiles/test-json-schema-to-grammar.dir/get-model.cpp.o
[ 53%] Linking CXX executable ../bin/test-json-schema-to-grammar
[ 53%] Built target test-json-schema-to-grammar
[ 54%] Building CXX object tests/CMakeFiles/test-quantize-stats.dir/test-quantize-stats.cpp.o
[ 54%] Linking CXX executable ../bin/test-quantize-stats
[ 54%] Built target test-quantize-stats
[ 54%] Building CXX object tests/CMakeFiles/test-gbnf-validator.dir/test-gbnf-validator.cpp.o
[ 54%] Linking CXX executable ../bin/test-gbnf-validator
[ 54%] Built target test-gbnf-validator
[ 54%] Building CXX object tests/CMakeFiles/test-tokenizer-1-bpe.dir/test-tokenizer-1-bpe.cpp.o
[ 55%] Linking CXX executable ../bin/test-tokenizer-1-bpe
[ 55%] Built target test-tokenizer-1-bpe
[ 55%] Building CXX object tests/CMakeFiles/test-tokenizer-1-spm.dir/test-tokenizer-1-spm.cpp.o
[ 55%] Linking CXX executable ../bin/test-tokenizer-1-spm
[ 55%] Built target test-tokenizer-1-spm
[ 55%] Building CXX object tests/CMakeFiles/test-chat-parser.dir/test-chat-parser.cpp.o
[ 55%] Building CXX object tests/CMakeFiles/test-chat-parser.dir/get-model.cpp.o
[ 56%] Linking CXX executable ../bin/test-chat-parser
[ 56%] Built target test-chat-parser
[ 56%] Building CXX object tests/CMakeFiles/test-chat-template.dir/test-chat-template.cpp.o
[ 56%] Building CXX object tests/CMakeFiles/test-chat-template.dir/get-model.cpp.o
[ 57%] Linking CXX executable ../bin/test-chat-template
[ 57%] Built target test-chat-template
[ 57%] Building CXX object tests/CMakeFiles/test-json-partial.dir/test-json-partial.cpp.o
[ 57%] Building CXX object tests/CMakeFiles/test-json-partial.dir/get-model.cpp.o
[ 58%] Linking CXX executable ../bin/test-json-partial
[ 58%] Built target test-json-partial
[ 59%] Building CXX object tests/CMakeFiles/test-log.dir/test-log.cpp.o
[ 59%] Building CXX object tests/CMakeFiles/test-log.dir/get-model.cpp.o
[ 59%] Linking CXX executable ../bin/test-log
[ 59%] Built target test-log
[ 59%] Building CXX object tests/CMakeFiles/test-regex-partial.dir/test-regex-partial.cpp.o
[ 60%] Building CXX object tests/CMakeFiles/test-regex-partial.dir/get-model.cpp.o
[ 60%] Linking CXX executable ../bin/test-regex-partial
[ 60%] Built target test-regex-partial
[ 60%] Building CXX object tests/CMakeFiles/test-thread-safety.dir/test-thread-safety.cpp.o
[ 60%] Building CXX object tests/CMakeFiles/test-thread-safety.dir/get-model.cpp.o
[ 60%] Linking CXX executable ../bin/test-thread-safety
[ 60%] Built target test-thread-safety
[ 60%] Building CXX object tests/CMakeFiles/test-arg-parser.dir/test-arg-parser.cpp.o
[ 61%] Building CXX object tests/CMakeFiles/test-arg-parser.dir/get-model.cpp.o
[ 61%] Linking CXX executable ../bin/test-arg-parser
[ 61%] Built target test-arg-parser
[ 61%] Building CXX object tests/CMakeFiles/test-opt.dir/test-opt.cpp.o
[ 61%] Building CXX object tests/CMakeFiles/test-opt.dir/get-model.cpp.o
[ 62%] Linking CXX executable ../bin/test-opt
[ 62%] Built target test-opt
[ 62%] Building CXX object tests/CMakeFiles/test-gguf.dir/test-gguf.cpp.o
[ 63%] Building CXX object tests/CMakeFiles/test-gguf.dir/get-model.cpp.o
[ 63%] Linking CXX executable ../bin/test-gguf
[ 63%] Built target test-gguf
[ 63%] Building CXX object tests/CMakeFiles/test-backend-ops.dir/test-backend-ops.cpp.o
[ 63%] Building CXX object tests/CMakeFiles/test-backend-ops.dir/get-model.cpp.o
[ 64%] Linking CXX executable ../bin/test-backend-ops
[ 64%] Built target test-backend-ops
[ 64%] Building CXX object tests/CMakeFiles/test-model-load-cancel.dir/test-model-load-cancel.cpp.o
[ 65%] Building CXX object tests/CMakeFiles/test-model-load-cancel.dir/get-model.cpp.o
[ 65%] Linking CXX executable ../bin/test-model-load-cancel
[ 65%] Built target test-model-load-cancel
[ 65%] Building CXX object tests/CMakeFiles/test-autorelease.dir/test-autorelease.cpp.o
[ 65%] Building CXX object tests/CMakeFiles/test-autorelease.dir/get-model.cpp.o
[ 66%] Linking CXX executable ../bin/test-autorelease
[ 66%] Built target test-autorelease
[ 66%] Building CXX object tests/CMakeFiles/test-barrier.dir/test-barrier.cpp.o
[ 66%] Building CXX object tests/CMakeFiles/test-barrier.dir/get-model.cpp.o
[ 66%] Linking CXX executable ../bin/test-barrier
[ 66%] Built target test-barrier
[ 66%] Building CXX object tests/CMakeFiles/test-quantize-fns.dir/test-quantize-fns.cpp.o
[ 66%] Building CXX object tests/CMakeFiles/test-quantize-fns.dir/get-model.cpp.o
[ 67%] Linking CXX executable ../bin/test-quantize-fns
[ 67%] Built target test-quantize-fns
[ 67%] Building CXX object tests/CMakeFiles/test-quantize-perf.dir/test-quantize-perf.cpp.o
[ 67%] Building CXX object tests/CMakeFiles/test-quantize-perf.dir/get-model.cpp.o
[ 67%] Linking CXX executable ../bin/test-quantize-perf
[ 67%] Built target test-quantize-perf
[ 67%] Building CXX object tests/CMakeFiles/test-rope.dir/test-rope.cpp.o
[ 67%] Building CXX object tests/CMakeFiles/test-rope.dir/get-model.cpp.o
[ 68%] Linking CXX executable ../bin/test-rope
[ 68%] Built target test-rope
[ 68%] Building CXX object tools/mtmd/CMakeFiles/mtmd.dir/mtmd.cpp.o
[ 69%] Building CXX object tools/mtmd/CMakeFiles/mtmd.dir/mtmd-audio.cpp.o
[ 69%] Building CXX object tools/mtmd/CMakeFiles/mtmd.dir/clip.cpp.o
[ 69%] Building CXX object tools/mtmd/CMakeFiles/mtmd.dir/mtmd-helper.cpp.o
[ 70%] Linking CXX shared library ../../bin/libmtmd.so
[ 70%] Built target mtmd
[ 70%] Building C object tests/CMakeFiles/test-mtmd-c-api.dir/test-mtmd-c-api.c.o
[ 71%] Building CXX object tests/CMakeFiles/test-mtmd-c-api.dir/get-model.cpp.o
[ 71%] Linking CXX executable ../bin/test-mtmd-c-api
[ 71%] Built target test-mtmd-c-api
[ 72%] Building C object tests/CMakeFiles/test-c.dir/test-c.c.o
[ 72%] Linking C executable ../bin/test-c
[ 72%] Built target test-c
[ 72%] Building CXX object tests/CMakeFiles/test-alloc.dir/test-alloc.cpp.o
[ 73%] Building CXX object tests/CMakeFiles/test-alloc.dir/get-model.cpp.o
[ 73%] Linking CXX executable ../bin/test-alloc
[ 73%] Built target test-alloc
[ 74%] Building CXX object examples/batched/CMakeFiles/llama-batched.dir/batched.cpp.o
[ 74%] Linking CXX executable ../../bin/llama-batched
[ 74%] Built target llama-batched
[ 74%] Building CXX object examples/embedding/CMakeFiles/llama-embedding.dir/embedding.cpp.o
[ 74%] Linking CXX executable ../../bin/llama-embedding
[ 74%] Built target llama-embedding
[ 74%] Building CXX object examples/eval-callback/CMakeFiles/llama-eval-callback.dir/eval-callback.cpp.o
[ 75%] Linking CXX executable ../../bin/llama-eval-callback
[ 75%] Built target llama-eval-callback
[ 75%] Building C object examples/gguf-hash/CMakeFiles/sha256.dir/deps/sha256/sha256.c.o
[ 75%] Built target sha256
[ 76%] Building C object examples/gguf-hash/CMakeFiles/xxhash.dir/deps/xxhash/xxhash.c.o
[ 76%] Built target xxhash
[ 76%] Building C object examples/gguf-hash/CMakeFiles/sha1.dir/deps/sha1/sha1.c.o
[ 76%] Built target sha1
[ 76%] Building CXX object examples/gguf-hash/CMakeFiles/llama-gguf-hash.dir/gguf-hash.cpp.o
[ 76%] Linking CXX executable ../../bin/llama-gguf-hash
[ 76%] Built target llama-gguf-hash
[ 76%] Building CXX object examples/gguf/CMakeFiles/llama-gguf.dir/gguf.cpp.o
[ 77%] Linking CXX executable ../../bin/llama-gguf
[ 77%] Built target llama-gguf
[ 78%] Building CXX object examples/lookahead/CMakeFiles/llama-lookahead.dir/lookahead.cpp.o
[ 78%] Linking CXX executable ../../bin/llama-lookahead
[ 78%] Built target llama-lookahead
[ 78%] Building CXX object examples/lookup/CMakeFiles/llama-lookup.dir/lookup.cpp.o
[ 79%] Linking CXX executable ../../bin/llama-lookup
[ 79%] Built target llama-lookup
[ 79%] Building CXX object examples/lookup/CMakeFiles/llama-lookup-create.dir/lookup-create.cpp.o
[ 79%] Linking CXX executable ../../bin/llama-lookup-create
[ 79%] Built target llama-lookup-create
[ 79%] Building CXX object examples/lookup/CMakeFiles/llama-lookup-merge.dir/lookup-merge.cpp.o
[ 80%] Linking CXX executable ../../bin/llama-lookup-merge
[ 80%] Built target llama-lookup-merge
[ 80%] Building CXX object examples/lookup/CMakeFiles/llama-lookup-stats.dir/lookup-stats.cpp.o
[ 80%] Linking CXX executable ../../bin/llama-lookup-stats
[ 80%] Built target llama-lookup-stats
[ 81%] Building CXX object examples/parallel/CMakeFiles/llama-parallel.dir/parallel.cpp.o
[ 81%] Linking CXX executable ../../bin/llama-parallel
[ 81%] Built target llama-parallel
[ 81%] Building CXX object examples/passkey/CMakeFiles/llama-passkey.dir/passkey.cpp.o
[ 82%] Linking CXX executable ../../bin/llama-passkey
[ 82%] Built target llama-passkey
[ 82%] Building CXX object examples/retrieval/CMakeFiles/llama-retrieval.dir/retrieval.cpp.o
[ 82%] Linking CXX executable ../../bin/llama-retrieval
[ 82%] Built target llama-retrieval
[ 83%] Building CXX object examples/save-load-state/CMakeFiles/llama-save-load-state.dir/save-load-state.cpp.o
[ 83%] Linking CXX executable ../../bin/llama-save-load-state
[ 83%] Built target llama-save-load-state
[ 83%] Building CXX object examples/simple/CMakeFiles/llama-simple.dir/simple.cpp.o
[ 84%] Linking CXX executable ../../bin/llama-simple
[ 84%] Built target llama-simple
[ 84%] Building CXX object examples/simple-chat/CMakeFiles/llama-simple-chat.dir/simple-chat.cpp.o
[ 84%] Linking CXX executable ../../bin/llama-simple-chat
[ 84%] Built target llama-simple-chat
[ 84%] Building CXX object examples/speculative/CMakeFiles/llama-speculative.dir/speculative.cpp.o
[ 85%] Linking CXX executable ../../bin/llama-speculative
[ 85%] Built target llama-speculative
[ 85%] Building CXX object examples/speculative-simple/CMakeFiles/llama-speculative-simple.dir/speculative-simple.cpp.o
[ 85%] Linking CXX executable ../../bin/llama-speculative-simple
[ 85%] Built target llama-speculative-simple
[ 86%] Building CXX object examples/gen-docs/CMakeFiles/llama-gen-docs.dir/gen-docs.cpp.o
[ 86%] Linking CXX executable ../../bin/llama-gen-docs
[ 86%] Built target llama-gen-docs
[ 87%] Building CXX object examples/training/CMakeFiles/llama-finetune.dir/finetune.cpp.o
[ 87%] Linking CXX executable ../../bin/llama-finetune
[ 87%] Built target llama-finetune
[ 87%] Building CXX object examples/diffusion/CMakeFiles/llama-diffusion-cli.dir/diffusion-cli.cpp.o
[ 88%] Linking CXX executable ../../bin/llama-diffusion-cli
[ 88%] Built target llama-diffusion-cli
[ 88%] Building CXX object examples/model-conversion/CMakeFiles/llama-logits.dir/logits.cpp.o
[ 88%] Linking CXX executable ../../bin/llama-logits
[ 88%] Built target llama-logits
[ 88%] Building CXX object examples/convert-llama2c-to-ggml/CMakeFiles/llama-convert-llama2c-to-ggml.dir/convert-llama2c-to-ggml.cpp.o
[ 88%] Linking CXX executable ../../bin/llama-convert-llama2c-to-ggml
[ 88%] Built target llama-convert-llama2c-to-ggml
[ 89%] Building CXX object pocs/vdot/CMakeFiles/llama-vdot.dir/vdot.cpp.o
[ 89%] Linking CXX executable ../../bin/llama-vdot
[ 89%] Built target llama-vdot
[ 89%] Building CXX object pocs/vdot/CMakeFiles/llama-q8dot.dir/q8dot.cpp.o
[ 90%] Linking CXX executable ../../bin/llama-q8dot
[ 90%] Built target llama-q8dot
[ 90%] Building CXX object tools/batched-bench/CMakeFiles/llama-batched-bench.dir/batched-bench.cpp.o
[ 91%] Linking CXX executable ../../bin/llama-batched-bench
[ 91%] Built target llama-batched-bench
[ 91%] Building CXX object tools/gguf-split/CMakeFiles/llama-gguf-split.dir/gguf-split.cpp.o
[ 92%] Linking CXX executable ../../bin/llama-gguf-split
[ 92%] Built target llama-gguf-split
[ 92%] Building CXX object tools/imatrix/CMakeFiles/llama-imatrix.dir/imatrix.cpp.o
[ 92%] Linking CXX executable ../../bin/llama-imatrix
[ 92%] Built target llama-imatrix
[ 92%] Building CXX object tools/llama-bench/CMakeFiles/llama-bench.dir/llama-bench.cpp.o
[ 92%] Linking CXX executable ../../bin/llama-bench
[ 92%] Built target llama-bench
[ 93%] Building CXX object tools/main/CMakeFiles/llama-cli.dir/main.cpp.o
[ 93%] Linking CXX executable ../../bin/llama-cli
[ 93%] Built target llama-cli
[ 93%] Building CXX object tools/perplexity/CMakeFiles/llama-perplexity.dir/perplexity.cpp.o
[ 93%] Linking CXX executable ../../bin/llama-perplexity
[ 93%] Built target llama-perplexity
[ 93%] Building CXX object tools/quantize/CMakeFiles/llama-quantize.dir/quantize.cpp.o
[ 93%] Linking CXX executable ../../bin/llama-quantize
[ 93%] Built target llama-quantize
[ 93%] Generating loading.html.hpp
[ 93%] Generating index.html.gz.hpp
[ 94%] Building CXX object tools/server/CMakeFiles/llama-server.dir/server.cpp.o
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4434:71: error: 'void res_error(httplib::Response&, const json&)' redeclared as different kind of entity
 4434 | static void res_error(httplib::Response & res, const json & error_data) {
      |                                                                       ^
In file included from /home/sagemaker-user/llama.cpp/common/../vendor/cpp-httplib/httplib.h:259,
                 from /home/sagemaker-user/llama.cpp/tools/server/utils.hpp:12,
                 from /home/sagemaker-user/llama.cpp/tools/server/server.cpp:2:
/opt/conda/x86_64-conda-linux-gnu/sysroot/usr/include/resolv.h:71:65: note: previous declaration 'res_sendhookact res_error'
   71 | typedef enum { res_goahead, res_nextns, res_modified, res_done, res_error }
      |                                                                 ^~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4527:22: error: 'res_error' cannot be used as a function
 4527 |             res_error(res, formatted_error);
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4535:22: error: 'res_error' cannot be used as a function
 4535 |             res_error(res, format_error_response("File Not Found", ERROR_TYPE_NOT_FOUND));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4594:18: error: 'res_error' cannot be used as a function
 4594 |         res_error(res, format_error_response("Invalid API Key", ERROR_TYPE_AUTHENTICATION));
      |         ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4612:26: error: 'res_error' cannot be used as a function
 4612 |                 res_error(res, format_error_response("Loading model", ERROR_TYPE_UNAVAILABLE));
      |                 ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4651:22: error: 'res_error' cannot be used as a function
 4651 |             res_error(res, format_error_response("This server does not support slots endpoint. Start it with `--slots`", ERROR_TYPE_NOT_SUPPORTED));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4669:22: error: 'res_error' cannot be used as a function
 4669 |             res_error(res, result->to_json());
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4680:26: error: 'res_error' cannot be used as a function
 4680 |                 res_error(res, format_error_response("no slot available", ERROR_TYPE_UNAVAILABLE));
      |                 ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4690:22: error: 'res_error' cannot be used as a function
 4690 |             res_error(res, format_error_response("This server does not support metrics endpoint. Start it with `--metrics`", ERROR_TYPE_NOT_SUPPORTED));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4708:22: error: 'res_error' cannot be used as a function
 4708 |             res_error(res, result->to_json());
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4793:22: error: 'res_error' cannot be used as a function
 4793 |             res_error(res, format_error_response("Invalid filename", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4814:22: error: 'res_error' cannot be used as a function
 4814 |             res_error(res, result->to_json());
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4825:22: error: 'res_error' cannot be used as a function
 4825 |             res_error(res, format_error_response("Invalid filename", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4846:22: error: 'res_error' cannot be used as a function
 4846 |             res_error(res, result->to_json());
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4869:22: error: 'res_error' cannot be used as a function
 4869 |             res_error(res, result->to_json());
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4879:22: error: 'res_error' cannot be used as a function
 4879 |             res_error(res, format_error_response("This server does not support slots action. Start it with `--slot-save-path`", ERROR_TYPE_NOT_SUPPORTED));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4889:22: error: 'res_error' cannot be used as a function
 4889 |             res_error(res, format_error_response("Invalid slot ID", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4902:22: error: 'res_error' cannot be used as a function
 4902 |             res_error(res, format_error_response("Invalid action", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:4950:22: error: 'res_error' cannot be used as a function
 4950 |             res_error(res, format_error_response("This server does not support changing global properties. Start it with `--props`", ERROR_TYPE_NOT_SUPPORTED));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5047:22: error: 'res_error' cannot be used as a function
 5047 |             res_error(res, format_error_response(e.what(), ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5059:26: error: 'res_error' cannot be used as a function
 5059 |                 res_error(res, all_results.error->to_json());
      |                 ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5079:26: error: 'res_error' cannot be used as a function
 5079 |                 res_error(res, first_result->to_json());
      |                 ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5186:22: error: 'res_error' cannot be used as a function
 5186 |             res_error(res, format_error_response(string_format("Infill is not supported by this model: %s", err.c_str()), ERROR_TYPE_NOT_SUPPORTED));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5195:22: error: 'res_error' cannot be used as a function
 5195 |             res_error(res, format_error_response("\"prompt\" must be a string", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5199:22: error: 'res_error' cannot be used as a function
 5199 |             res_error(res, format_error_response("\"input_prefix\" is required", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5203:22: error: 'res_error' cannot be used as a function
 5203 |             res_error(res, format_error_response("\"input_suffix\" is required", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5208:22: error: 'res_error' cannot be used as a function
 5208 |             res_error(res, format_error_response("\"input_extra\" must be an array of {\"filename\": string, \"text\": string}", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5216:26: error: 'res_error' cannot be used as a function
 5216 |                 res_error(res, format_error_response("extra_context chunk must contain a \"text\" field with a string value", ERROR_TYPE_INVALID_REQUEST));
      |                 ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5221:26: error: 'res_error' cannot be used as a function
 5221 |                 res_error(res, format_error_response("extra_context chunk's \"filename\" field must be a string", ERROR_TYPE_INVALID_REQUEST));
      |                 ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5383:22: error: 'res_error' cannot be used as a function
 5383 |             res_error(res, format_error_response("This server does not support embeddings. Start it with `--embeddings`", ERROR_TYPE_NOT_SUPPORTED));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5388:22: error: 'res_error' cannot be used as a function
 5388 |             res_error(res, format_error_response("Pooling type 'none' is not OAI compatible. Please use a different pooling type", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5402:22: error: 'res_error' cannot be used as a function
 5402 |             res_error(res, format_error_response("\"input\" or \"content\" must be provided", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5412:26: error: 'res_error' cannot be used as a function
 5412 |                 res_error(res, format_error_response("The format to return the embeddings in. Can be either float or base64", ERROR_TYPE_INVALID_REQUEST));
      |                 ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5421:26: error: 'res_error' cannot be used as a function
 5421 |                 res_error(res, format_error_response("Input content cannot be empty", ERROR_TYPE_INVALID_REQUEST));
      |                 ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5462:22: error: 'res_error' cannot be used as a function
 5462 |             res_error(res, all_results.error->to_json());
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5488:22: error: 'res_error' cannot be used as a function
 5488 |             res_error(res, format_error_response("This server does not support reranking. Start it with `--reranking`", ERROR_TYPE_NOT_SUPPORTED));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5503:26: error: 'res_error' cannot be used as a function
 5503 |                 res_error(res, format_error_response("\"query\" must be a string", ERROR_TYPE_INVALID_REQUEST));
      |                 ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5507:22: error: 'res_error' cannot be used as a function
 5507 |             res_error(res, format_error_response("\"query\" must be provided", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5514:22: error: 'res_error' cannot be used as a function
 5514 |             res_error(res, format_error_response("\"documents\" must be a non-empty string array", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5544:22: error: 'res_error' cannot be used as a function
 5544 |             res_error(res, all_results.error->to_json());
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp: In lambda function:
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5597:22: error: 'res_error' cannot be used as a function
 5597 |             res_error(res, format_error_response("Request body must be an array", ERROR_TYPE_INVALID_REQUEST));
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/sagemaker-user/llama.cpp/tools/server/server.cpp:5615:22: error: 'res_error' cannot be used as a function
 5615 |             res_error(res, result->to_json());
      |             ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
make[2]: *** [tools/server/CMakeFiles/llama-server.dir/build.make:84: tools/server/CMakeFiles/llama-server.dir/server.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:3909: tools/server/CMakeFiles/llama-server.dir/all] Error 2
make: *** [Makefile:146: all] Error 2
```

--- a/tools/server/server.cpp
+++ b/tools/server/server.cpp
@@ -4431,7 +4431,7 @@ static void log_server_request(const httplib::Request & req, const httplib::Resp
     SRV_DBG("response: %s\n", res.body.c_str());
 }
 
-static void res_error(httplib::Response & res, const json & error_data) {
+static void res_err(httplib::Response & res, const json & error_data) {
     json final_response {{"error", error_data}};
     res.set_content(safe_json_to_str(final_response), MIMETYPE_JSON);
     res.status = json_value(error_data, "code", 500);
@@ -4524,7 +4524,7 @@ int main(int argc, char ** argv) {
         try {
             json formatted_error = format_error_response(message, ERROR_TYPE_SERVER);
             LOG_WRN("got exception: %s\n", formatted_error.dump().c_str());
-            res_error(res, formatted_error);
+            res_err(res, formatted_error);
         } catch (const std::exception & e) {
             LOG_ERR("got another exception: %s | while hanlding exception: %s\n", e.what(), message.c_str());
         }
@@ -4532,9 +4532,9 @@ int main(int argc, char ** argv) {
 
     svr->set_error_handler([](const httplib::Request &, httplib::Response & res) {
         if (res.status == 404) {
-            res_error(res, format_error_response("File Not Found", ERROR_TYPE_NOT_FOUND));
+            res_err(res, format_error_response("File Not Found", ERROR_TYPE_NOT_FOUND));
         }
-        // for other error codes, we skip processing here because it's already done by res_error()
+        // for other error codes, we skip processing here because it's already done by res_err()
     });
 
     // set timeouts and change hostname and port
@@ -4591,7 +4591,7 @@ int main(int argc, char ** argv) {
         }
 
         // API key is invalid or not provided
-        res_error(res, format_error_response("Invalid API Key", ERROR_TYPE_AUTHENTICATION));
+        res_err(res, format_error_response("Invalid API Key", ERROR_TYPE_AUTHENTICATION));
 
         LOG_WRN("Unauthorized: Invalid API Key\n");
 
@@ -4609,7 +4609,7 @@ int main(int argc, char ** argv) {
                 // allow the models endpoint to be accessed during loading
                 return true;
             } else {
-                res_error(res, format_error_response("Loading model", ERROR_TYPE_UNAVAILABLE));
+                res_err(res, format_error_response("Loading model", ERROR_TYPE_UNAVAILABLE));
             }
             return false;
         }
@@ -4648,7 +4648,7 @@ int main(int argc, char ** argv) {
 
     const auto handle_slots = [&](const httplib::Request & req, httplib::Response & res) {
         if (!params.endpoint_slots) {
-            res_error(res, format_error_response("This server does not support slots endpoint. Start it with `--slots`", ERROR_TYPE_NOT_SUPPORTED));
+            res_err(res, format_error_response("This server does not support slots endpoint. Start it with `--slots`", ERROR_TYPE_NOT_SUPPORTED));
             return;
         }
 
@@ -4666,7 +4666,7 @@ int main(int argc, char ** argv) {
         ctx_server.queue_results.remove_waiting_task_id(task_id);
 
         if (result->is_error()) {
-            res_error(res, result->to_json());
+            res_err(res, result->to_json());
             return;
         }
 
@@ -4677,7 +4677,7 @@ int main(int argc, char ** argv) {
         // optionally return "fail_on_no_slot" error
         if (req.has_param("fail_on_no_slot")) {
             if (res_task->n_idle_slots == 0) {
-                res_error(res, format_error_response("no slot available", ERROR_TYPE_UNAVAILABLE));
+                res_err(res, format_error_response("no slot available", ERROR_TYPE_UNAVAILABLE));
                 return;
             }
         }
@@ -4687,7 +4687,7 @@ int main(int argc, char ** argv) {
 
     const auto handle_metrics = [&](const httplib::Request &, httplib::Response & res) {
         if (!params.endpoint_metrics) {
-            res_error(res, format_error_response("This server does not support metrics endpoint. Start it with `--metrics`", ERROR_TYPE_NOT_SUPPORTED));
+            res_err(res, format_error_response("This server does not support metrics endpoint. Start it with `--metrics`", ERROR_TYPE_NOT_SUPPORTED));
             return;
         }
 
@@ -4705,7 +4705,7 @@ int main(int argc, char ** argv) {
         ctx_server.queue_results.remove_waiting_task_id(task_id);
 
         if (result->is_error()) {
-            res_error(res, result->to_json());
+            res_err(res, result->to_json());
             return;
         }
 
@@ -4790,7 +4790,7 @@ int main(int argc, char ** argv) {
         json request_data = json::parse(req.body);
         std::string filename = request_data.at("filename");
         if (!fs_validate_filename(filename)) {
-            res_error(res, format_error_response("Invalid filename", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("Invalid filename", ERROR_TYPE_INVALID_REQUEST));
             return;
         }
         std::string filepath = params.slot_save_path + filename;
@@ -4811,7 +4811,7 @@ int main(int argc, char ** argv) {
         ctx_server.queue_results.remove_waiting_task_id(task_id);
 
         if (result->is_error()) {
-            res_error(res, result->to_json());
+            res_err(res, result->to_json());
             return;
         }
 
@@ -4822,7 +4822,7 @@ int main(int argc, char ** argv) {
         json request_data = json::parse(req.body);
         std::string filename = request_data.at("filename");
         if (!fs_validate_filename(filename)) {
-            res_error(res, format_error_response("Invalid filename", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("Invalid filename", ERROR_TYPE_INVALID_REQUEST));
             return;
         }
         std::string filepath = params.slot_save_path + filename;
@@ -4843,7 +4843,7 @@ int main(int argc, char ** argv) {
         ctx_server.queue_results.remove_waiting_task_id(task_id);
 
         if (result->is_error()) {
-            res_error(res, result->to_json());
+            res_err(res, result->to_json());
             return;
         }
 
@@ -4866,7 +4866,7 @@ int main(int argc, char ** argv) {
         ctx_server.queue_results.remove_waiting_task_id(task_id);
 
         if (result->is_error()) {
-            res_error(res, result->to_json());
+            res_err(res, result->to_json());
             return;
         }
 
@@ -4876,7 +4876,7 @@ int main(int argc, char ** argv) {
 
     const auto handle_slots_action = [&params, &handle_slots_save, &handle_slots_restore, &handle_slots_erase](const httplib::Request & req, httplib::Response & res) {
         if (params.slot_save_path.empty()) {
-            res_error(res, format_error_response("This server does not support slots action. Start it with `--slot-save-path`", ERROR_TYPE_NOT_SUPPORTED));
+            res_err(res, format_error_response("This server does not support slots action. Start it with `--slot-save-path`", ERROR_TYPE_NOT_SUPPORTED));
             return;
         }
 
@@ -4886,7 +4886,7 @@ int main(int argc, char ** argv) {
         try {
             id_slot = std::stoi(id_slot_str);
         } catch (const std::exception &) {
-            res_error(res, format_error_response("Invalid slot ID", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("Invalid slot ID", ERROR_TYPE_INVALID_REQUEST));
             return;
         }
 
@@ -4899,7 +4899,7 @@ int main(int argc, char ** argv) {
         } else if (action == "erase") {
             handle_slots_erase(req, res, id_slot);
         } else {
-            res_error(res, format_error_response("Invalid action", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("Invalid action", ERROR_TYPE_INVALID_REQUEST));
         }
     };
 
@@ -4947,7 +4947,7 @@ int main(int argc, char ** argv) {
 
     const auto handle_props_change = [&ctx_server](const httplib::Request & req, httplib::Response & res) {
         if (!ctx_server.params_base.endpoint_props) {
-            res_error(res, format_error_response("This server does not support changing global properties. Start it with `--props`", ERROR_TYPE_NOT_SUPPORTED));
+            res_err(res, format_error_response("This server does not support changing global properties. Start it with `--props`", ERROR_TYPE_NOT_SUPPORTED));
             return;
         }
 
@@ -5044,7 +5044,7 @@ int main(int argc, char ** argv) {
 
             rd->post_tasks(std::move(tasks));
         } catch (const std::exception & e) {
-            res_error(res, format_error_response(e.what(), ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response(e.what(), ERROR_TYPE_INVALID_REQUEST));
             return;
         }
 
@@ -5056,7 +5056,7 @@ int main(int argc, char ** argv) {
             if (all_results.is_terminated) {
                 return; // connection is closed
             } else if (all_results.error) {
-                res_error(res, all_results.error->to_json());
+                res_err(res, all_results.error->to_json());
                 return;
             } else {
                 json arr = json::array();
@@ -5076,7 +5076,7 @@ int main(int argc, char ** argv) {
             if (first_result == nullptr) {
                 return; // connection is closed
             } else if (first_result->is_error()) {
-                res_error(res, first_result->to_json());
+                res_err(res, first_result->to_json());
                 return;
             } else {
                 GGML_ASSERT(
@@ -5183,7 +5183,7 @@ int main(int argc, char ** argv) {
             err += "middle token is missing. ";
         }
         if (!err.empty()) {
-            res_error(res, format_error_response(string_format("Infill is not supported by this model: %s", err.c_str()), ERROR_TYPE_NOT_SUPPORTED));
+            res_err(res, format_error_response(string_format("Infill is not supported by this model: %s", err.c_str()), ERROR_TYPE_NOT_SUPPORTED));
             return;
         }
 
@@ -5192,20 +5192,20 @@ int main(int argc, char ** argv) {
         // validate input
         if (data.contains("prompt") && !data.at("prompt").is_string()) {
             // prompt is optional
-            res_error(res, format_error_response("\"prompt\" must be a string", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("\"prompt\" must be a string", ERROR_TYPE_INVALID_REQUEST));
         }
 
         if (!data.contains("input_prefix")) {
-            res_error(res, format_error_response("\"input_prefix\" is required", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("\"input_prefix\" is required", ERROR_TYPE_INVALID_REQUEST));
         }
 
         if (!data.contains("input_suffix")) {
-            res_error(res, format_error_response("\"input_suffix\" is required", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("\"input_suffix\" is required", ERROR_TYPE_INVALID_REQUEST));
         }
 
         if (data.contains("input_extra") && !data.at("input_extra").is_array()) {
             // input_extra is optional
-            res_error(res, format_error_response("\"input_extra\" must be an array of {\"filename\": string, \"text\": string}", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("\"input_extra\" must be an array of {\"filename\": string, \"text\": string}", ERROR_TYPE_INVALID_REQUEST));
             return;
         }
 
@@ -5213,12 +5213,12 @@ int main(int argc, char ** argv) {
         for (const auto & chunk : input_extra) {
             // { "text": string, "filename": string }
             if (!chunk.contains("text") || !chunk.at("text").is_string()) {
-                res_error(res, format_error_response("extra_context chunk must contain a \"text\" field with a string value", ERROR_TYPE_INVALID_REQUEST));
+                res_err(res, format_error_response("extra_context chunk must contain a \"text\" field with a string value", ERROR_TYPE_INVALID_REQUEST));
                 return;
             }
             // filename is optional
             if (chunk.contains("filename") && !chunk.at("filename").is_string()) {
-                res_error(res, format_error_response("extra_context chunk's \"filename\" field must be a string", ERROR_TYPE_INVALID_REQUEST));
+                res_err(res, format_error_response("extra_context chunk's \"filename\" field must be a string", ERROR_TYPE_INVALID_REQUEST));
                 return;
             }
         }
@@ -5380,12 +5380,12 @@ int main(int argc, char ** argv) {
 
     const auto handle_embeddings_impl = [&ctx_server](const httplib::Request & req, httplib::Response & res, oaicompat_type oaicompat) {
         if (!ctx_server.params_base.embedding) {
-            res_error(res, format_error_response("This server does not support embeddings. Start it with `--embeddings`", ERROR_TYPE_NOT_SUPPORTED));
+            res_err(res, format_error_response("This server does not support embeddings. Start it with `--embeddings`", ERROR_TYPE_NOT_SUPPORTED));
             return;
         }
 
         if (oaicompat != OAICOMPAT_TYPE_NONE && llama_pooling_type(ctx_server.ctx) == LLAMA_POOLING_TYPE_NONE) {
-            res_error(res, format_error_response("Pooling type 'none' is not OAI compatible. Please use a different pooling type", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("Pooling type 'none' is not OAI compatible. Please use a different pooling type", ERROR_TYPE_INVALID_REQUEST));
             return;
         }
 
@@ -5399,7 +5399,7 @@ int main(int argc, char ** argv) {
             oaicompat = OAICOMPAT_TYPE_NONE; // "content" field is not OAI compatible
             prompt = body.at("content");
         } else {
-            res_error(res, format_error_response("\"input\" or \"content\" must be provided", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("\"input\" or \"content\" must be provided", ERROR_TYPE_INVALID_REQUEST));
             return;
         }
 
@@ -5409,7 +5409,7 @@ int main(int argc, char ** argv) {
             if (format == "base64") {
                 use_base64 = true;
             } else if (format != "float") {
-                res_error(res, format_error_response("The format to return the embeddings in. Can be either float or base64", ERROR_TYPE_INVALID_REQUEST));
+                res_err(res, format_error_response("The format to return the embeddings in. Can be either float or base64", ERROR_TYPE_INVALID_REQUEST));
                 return;
             }
         }
@@ -5418,7 +5418,7 @@ int main(int argc, char ** argv) {
         for (const auto & tokens : tokenized_prompts) {
             // this check is necessary for models that do not add BOS token to the input
             if (tokens.empty()) {
-                res_error(res, format_error_response("Input content cannot be empty", ERROR_TYPE_INVALID_REQUEST));
+                res_err(res, format_error_response("Input content cannot be empty", ERROR_TYPE_INVALID_REQUEST));
                 return;
             }
         }
@@ -5459,7 +5459,7 @@ int main(int argc, char ** argv) {
         if (all_results.is_terminated) {
             return; // connection is closed
         } else if (all_results.error) {
-            res_error(res, all_results.error->to_json());
+            res_err(res, all_results.error->to_json());
             return;
         } else {
             for (auto & res : all_results.results) {
@@ -5485,7 +5485,7 @@ int main(int argc, char ** argv) {
 
     const auto handle_rerank = [&ctx_server](const httplib::Request & req, httplib::Response & res) {
         if (!ctx_server.params_base.embedding || ctx_server.params_base.pooling_type != LLAMA_POOLING_TYPE_RANK) {
-            res_error(res, format_error_response("This server does not support reranking. Start it with `--reranking`", ERROR_TYPE_NOT_SUPPORTED));
+            res_err(res, format_error_response("This server does not support reranking. Start it with `--reranking`", ERROR_TYPE_NOT_SUPPORTED));
             return;
         }
 
@@ -5500,18 +5500,18 @@ int main(int argc, char ** argv) {
         if (body.count("query") == 1) {
             query = body.at("query");
             if (!query.is_string()) {
-                res_error(res, format_error_response("\"query\" must be a string", ERROR_TYPE_INVALID_REQUEST));
+                res_err(res, format_error_response("\"query\" must be a string", ERROR_TYPE_INVALID_REQUEST));
                 return;
             }
         } else {
-            res_error(res, format_error_response("\"query\" must be provided", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("\"query\" must be provided", ERROR_TYPE_INVALID_REQUEST));
             return;
         }
 
         std::vector<std::string> documents = json_value(body, "documents",
                                              json_value(body, "texts", std::vector<std::string>()));
         if (documents.empty()) {
-            res_error(res, format_error_response("\"documents\" must be a non-empty string array", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("\"documents\" must be a non-empty string array", ERROR_TYPE_INVALID_REQUEST));
             return;
         }
 
@@ -5541,7 +5541,7 @@ int main(int argc, char ** argv) {
         if (all_results.is_terminated) {
             return; // connection is closed
         } else if (all_results.error) {
-            res_error(res, all_results.error->to_json());
+            res_err(res, all_results.error->to_json());
             return;
         } else {
             for (auto & res : all_results.results) {
@@ -5594,7 +5594,7 @@ int main(int argc, char ** argv) {
     const auto handle_lora_adapters_apply = [&](const httplib::Request & req, httplib::Response & res) {
         const json body = json::parse(req.body);
         if (!body.is_array()) {
-            res_error(res, format_error_response("Request body must be an array", ERROR_TYPE_INVALID_REQUEST));
+            res_err(res, format_error_response("Request body must be an array", ERROR_TYPE_INVALID_REQUEST));
             return;
         }
 
@@ -5612,7 +5612,7 @@ int main(int argc, char ** argv) {
         ctx_server.queue_results.remove_waiting_task_id(task_id);
 
         if (result->is_error()) {
-            res_error(res, result->to_json());
+            res_err(res, result->to_json());
             return;
         }
 

